src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java

```diff
@@ -280,6 +286,12 @@ private void ingestFunctionGroups(String externalServiceAgreementId, boolean isR
                         template.getJobProfileName(), isRetail);
                     return;
                 }
+                if(isTemplate.test(template)) {
+                    String serviceAgreementName = serviceAgreementsIntegrationRestClient
+                        .retrieveServiceAgreementByExternalId(externalServiceAgreementId).getName();
+                    if(!isRootServiceAgreement.test(serviceAgreementName))
+                        return;
+                }
```

Comment:
- Missing space in the `if` statements
- `if` statement without curly brackets should be avoided

Reply:
Done